documentation/Tutorials/WPF/SimpleContainer.md

```diff
@@ -0,0 +1,116 @@
+---
+layout: page
+title: Add a simple container
+---
+
+[Contents](Contents) [Previous](Menu) [Next](DialogForm)
+
+## Dependency control container
+
+In this part of the WPF Tutorial an Inversion of Control Container will be added. This container is a kind of dispenser for class objects. You also will find the term Dependency Injection used often. You can do without and use new statements to instantiate class objects, but for complex applicationt using a container is considered best practice for good reasons. For more background, one of your options is to watch this video: [Tim Corey Dependency Inversion tutorial](https://www.youtube.com/watch?v=NnZZMkwI6KI)
+
+For this tutorial a Caliburn.Micro ``SimpleContainer`` will be used. Make following changes t ``Bootstrapper.cs``:
+
+Add an interface to the container:
+
+```C#
+private readonly SimpleContainer _container = new SimpleContainer();
+```
+
+There are quite a lot of different libraries to do similar things, and you can combine those libraries with Caliburn.Micro but this is a Caliburn.Micro tutorial and in many cases this will do the job.
+
+We need to setup the contents of the container by overriding the ``Configure`` method:
+
+```C#
+   protected override void Configure()
+     {
+     _container.Instance(_container);
+     _container
+       .Singleton<IWindowManager, WindowManager>()
+       .Singleton<IEventAggregator, EventAggregator>();
+
+     foreach(var assembly in SelectAssemblies())
```

Comment:
In the features samples we manually register the ViewModels instead of using reflection.  You should mention you can do it both ways.            container
               .PerRequest<ShellViewModel>()
               .PerRequest<MenuViewModel>()
               .PerRequest<BindingsViewModel>()
               .PerRequest<ActionsViewModel>()
               .PerRequest<CoroutineViewModel>()
               .PerRequest<ExecuteViewModel>()
               .PerRequest<EventAggregationViewModel>()
               .PerRequest<DesignTimeViewModel>()
               .PerRequest<ConductorViewModel>()
               .PerRequest<BubblingViewModel>()
               .PerRequest<NavigationSourceViewModel>()
               .PerRequest<NavigationTargetViewModel>();

            container
               .PerRequest<ShellViewModel>()
               .PerRequest<MenuViewModel>()
               .PerRequest<BindingsViewModel>()
               .PerRequest<ActionsViewModel>()
               .PerRequest<CoroutineViewModel>()
               .PerRequest<ExecuteViewModel>()
               .PerRequest<EventAggregationViewModel>()
               .PerRequest<DesignTimeViewModel>()
               .PerRequest<ConductorViewModel>()
               .PerRequest<BubblingViewModel>()
               .PerRequest<NavigationSourceViewModel>()
               .PerRequest<NavigationTargetViewModel>();

Reply:
Good comment. I added this way of working to the text.